state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -221,7 +227,7 @@ pub mod pallet {
 
 	/// Scheduled Swaps
 	#[pallet::storage]
-	pub(crate) type SwapQueue<T: Config> = StorageValue<_, Vec<Swap>, ValueQuery>;
+	pub(crate) type SwapQueue<T: Config> = StorageValue<_, Vec<Swap<T>>, ValueQuery>;
```

Comment:
I think it might be better to use a Map<BlockNumber, Vec<Swap<T>> (+ Value<BlockNumber> for the last block executed)...

This way we always have to decode all swaps (including future swaps) and then filter them out, and write the remaining Vec back to storage. It's wasteful. Also the Vec can grow arbitrarily large (and repeats the block_number value for every swap, which is a waste). Also keeps the Swap struct unchanged - no need for the `T` type param and the migration is much simpler.

Reply:
> I think it might be better to use a Map<BlockNumber, Vec<Swap> (+ Value for the last block executed)...

I did consider this, but I thought it'd be better to keep things simple? With the map we'd need to be more careful when selecting which keys to query (because of failures). It is not very hard, but more room for error still.

> This way we always have to decode all swaps (including future swaps) and then filter them out, and write the remaining Vec back to storage. It's wasteful. 

I don't have a good intiution when it comes to SC runtime's requirements for performance, but we are only talking about a handful of swaps per block, I imagine (e.g. looks like TC does 30 swaps per minute on average)?

I'm happy to make the change if you think it is necessary, but to me it seems like an optimisation we could consider in the future?

Reply:
Yeah, I was thinking about this approach too, but I did consider it an optimisation, given we already do query for every swap. I do agree it's not nice having the block number duplicated in every swap, but in reality, how often are we getting two swaps in a 6 second window? So imo it's worth tracking this (creating an issue) as a future optimisation (if we ever needed to optimise it at all.